build/Release.ps1

```diff
@@ -0,0 +1,156 @@
+param (
+    [Parameter(Mandatory)]
+    # https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+    [ValidatePattern("^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$")]
+    [string]
+    $TagName,
+    [Parameter(Mandatory)]
+    [ValidatePattern("[0-9a-z]{46}")]
+    [string]
+    $NugetApiKey,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [bool]
+    $CoreComponents,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [bool]
+    $NonCoreComponents,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [bool]
+    $IsStableRelease
+)
+
+Function PressKeyToContinue {
+    param (
+        $Instruction
+    )
+
+    Write-Host $Instruction
+    Write-Host -NoNewLine "Press any key to continue..."
+    $Null = $Host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")
+}
+
+Function GitFetchAndMergeUpstreamMain {
+    git fetch upstream main
```

Comment:
what happens if it fails? e.g. when one does not have `upstream` remote
I think we should check the exit code after such invokes
we can create a helper function e.g. like this (taken from PSake):

```pwsh
function Exec
{
<#
.SYNOPSIS
Helper function for executing command-line programs.
.DESCRIPTION
This is a helper function that runs a scriptblock and checks the PS variable $lastexitcode to see if an error occcured.
If an error is detected then an exception is thrown.  This function allows you to run command-line programs without
having to explicitly check fthe $lastexitcode variable.
.PARAMETER cmd
The scriptblock to execute.  This scriptblock will typically contain the command-line invocation.
Required
.PARAMETER errorMessage
The error message used for the exception that is thrown.
Optional
.EXAMPLE
exec { svn info $repository_trunk } "Error executing SVN. Please verify SVN command-line client is installed"
This example calls the svn command-line client.
TaskTearDown
#>
[CmdletBinding(
    SupportsShouldProcess=$False,
    SupportsTransactions=$False,
    ConfirmImpact="None",
    DefaultParameterSetName="")]

  param(
    [Parameter(Position=0,Mandatory=1)][scriptblock]$cmd,
    [Parameter(Position=1,Mandatory=0)][string]$errorMessage = "Error executing command: " + $cmd
  )
     & $cmd
     if ($lastexitcode -ne 0)
     {
          throw $errorMessage
     }
}
```